### PR TITLE
Fix setuptools sdist

### PR DESCRIPTION
**Pull request by charris**
Fix for #7127: Setuptools sdist doesn't include all the files in the extension sources list.

Also add test installing from sdist archive to TravisCI.

### CODE REVIEW

**Comment by rgommers** [comment]
Checked the build log of the ``USE_SDIST`` run on TravisCI, all looks good.

**Comment by njsmith** [comment]
Do we know why that test didn't catch this in the first place?

On Wed, Jan 27, 2016 at 2:12 PM, Ralf Gommers <notifications@github.com>
wrote:

> Checked the build log of the USE_SDIST run on TravisCI, all looks good.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/7131#issuecomment-175887487>.
>



-- 
Nathaniel J. Smith -- https://vorpus.org <http://vorpus.org>

**Comment by charris** [comment]
@njsmith The test installing from the sdist created archives is new. Don't know if that answers your question.

**Comment by njsmith** [comment]
Ah, right, sorry :-) I guess I misremembered us adding something like that
before...

On Wed, Jan 27, 2016 at 2:35 PM, Charles Harris <notifications@github.com>
wrote:

> Merged #7131 <https://github.com/numpy/numpy/pull/7131>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/7131#event-528884531>.
>



-- 
Nathaniel J. Smith -- https://vorpus.org <http://vorpus.org>

**Comment by rgommers** [comment]
> Ah, right, sorry :-) I guess I misremembered us adding something like that before...

The wheel tests were added fairly recently, but those don't exercise ``sdist`` (yet, I fear)

File: numpy/distutils/command/egg_info.py [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 
+import sys
+
 from setuptools.command.egg_info import egg_info as _egg_info
 
 class egg_info(_egg_info):
     def run(self):
+        if 'sdist' in sys.argv:
+            import warnings
+            warnings.warn("`build_src` is being run, this may lead to missing "
```
Comment (stefanv) [inline numpy/distutils/command/egg_info.py]: @charris @rgommers I ran into this issue with one of the skimage tools, which checks to see whether our MANIFEST file is complete.  It says to look at gh-7127, but I'm not sure what to look at!  Could we point to a likely fix in this message instead?  I am happy to make the patch, but was hoping you could point me in the right direction.

Comment (stefanv) [inline numpy/distutils/command/egg_info.py]: Here's the tool executing `python setup.py sdist`, for what it's worth: https://github.com/stefanv/scikit-image/blob/check_sdist/tools/check_sdist.py#L12

Comment (rgommers) [inline numpy/distutils/command/egg_info.py]: The fix is in this PR: 6770f985b3

Comment (rgommers) [inline numpy/distutils/command/egg_info.py]: In the ``skimage`` ``setup.py`` you don't have any ``sdist`` handling, so you also have to add something like ``cmdclass={'sdist': sdist}`` or (better) the ``sdist_checked`` version that numpy and scipy have.

Comment (stefanv) [inline numpy/distutils/command/egg_info.py]: Thanks, @rgommers, that's very helpful.

Comment (rgommers) [inline numpy/distutils/command/egg_info.py]: You're welcome. I'll try to get around to tweaking the warning message.

Comment (rgommers) [inline numpy/distutils/command/egg_info.py]: It took a while, but addressed in gh-8761

**Pull request merged**